api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -116,6 +118,21 @@ public static void set(OpenTelemetry openTelemetry) {
     }
   }
 
+  /**
+   * Sets the {@link OpenTelemetry} that should be the global instance.
+   *
+   * <p>This method calls the given {@code supplier}, converts the output to an {@link
+   * OpenTelemetry} object, and finally calls {@link #set(OpenTelemetry)}, all while holding the
+   * {@link GlobalOpenTelemetry} mutex.
+   */
+  public static <T> T set(Supplier<T> supplier, Function<T, OpenTelemetry> converter) {
```

Comment:
What about something like: https://github.com/open-telemetry/opentelemetry-java/commit/5f06de2b8aeca6a41157a88d3abaab3bd23dbded

This API signature will look strange outside of the autoconfigure use case.

I like this approach in general though and would love to get it merged!